instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,326 @@
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForOneSecondLongFreeDrive() = sdkTest {
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        val billingRequests = getAllBillingEventsWhenHasEvents(1, "Free Drive") {
+            it.isFreeDriveSession
+        }
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForEmptyFreeDrive() = sdkTest {
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        val billingRequests = getAllBillingEventsWhenHasEvents(1, "Free Drive") {
+            it.isFreeDriveSession
+        }
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForOneSecondLongPausedAndResumedFreeDrive() = sdkTest {
+        val sessionLengthMillis = TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS)
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(sessionLengthMillis / 2)
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(sessionLengthMillis / 2)
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        mapboxNavigation.onDestroy()
+
+        val billingRequests = getAllBillingEventsWhenHasEvents(1, "Free Drive") {
+            it.isFreeDriveSession
+        }
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionsForMixedFreeDriveAndActiveGuidanceTrip() = sdkTest {
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(1000)
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        delay(2000)
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+        delay(3000)
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        val billingRequests = getAllBillingEventsWhenHasEvents(
+            3,
+            "Free Drive and Active Guidance"
```

Comment:
NIT: "any" because you don't filter out Idle. They shouldn't be sent, but it's a test and we don't know anything about it.